pkg/distributor/distributor_test.go

```diff
@@ -2488,7 +2488,7 @@ func TestDistributor_IngestionIsControlledByForwarder(t *testing.T) {
 				getForwarder:      getForwarder,
 			})
 
-			wrappedPush := distributors[0].PrePushForwardingMiddleware(distributors[0].PushWithCleanup)
+			wrappedPush := distributors[0].prePushForwardingMiddleware(distributors[0].PushWithCleanup)
```

Comment:
I think this will be a more complete test if we use the push, which includes all middlewares
```suggestion
			wrappedPush := distributors[0].Push
			response, err := wrappedPush(ctx, tc.request)
```

Reply:
thx, addressed it here: [5bace6f](https://github.com/grafana/mimir/pull/2603/commits/5bace6f3c17ffdf9133ac2f833e9ede87c6547b3)